Feat/rollup html social urls (#838)
modernweb-dev/web: docs/docs/building/rollup-plugin-html.md

### Bundling assets

> The HTML plugin will bundle assets referenced from `img` and `link` and `meta property="og:image"` elements in your HTML. The assets are emitted as rollup assets, and the paths are updated to the rollup output paths.

Reviewer: ```suggestion
The HTML plugin will bundle assets referenced from `img` and `link` and social media tags in your HTML. The assets are emitted as rollup assets, and the paths are updated to the rollup output paths.
```

The code examples shows it handles more than just `og:image`

Author: changed

The HTML plugin does not do any minification by default. You can use a transform function to use a minifier for your HTML or assets.

### Social Media Tags

> Some tags/URLs do need to use full absolute URLs (e.g. https://domain.com/guide/).

Reviewer: ```suggestion
Some social media tags require full absolute URLs (e.g. https://domain.com/guide/).
```

Author: changed

  transformHtml?: TransformHtmlFunction | TransformHtmlFunction[];
  /** Whether to extract and bundle assets referenced in HTML. Defaults to true. */
  extractAssets?: boolean;
  /** Define a full absolute url to your site (e.g. https://domain.com) */

>   siteUrl?: string;

Reviewer: I don't think this is a great name. It should at least be website but not everything is a website. It's rather the deployed origin and base path... is there a common name for that?

Reviewer: What about just:

```suggestion
  baseUrl?: string;
```

Author: it could be a URL with subfolder as well... e.g. domain.com/my-sub-folder

so it's a full URL to a site/website...

siteUrl or websiteUrl... 
don't really have an idea about a different name 🤔

Author: used `absoluteBaseUrl`

  /** Whether to set full absolute urls for ['meta[property=og:image]', 'link[rel=canonical]', 'meta[property=og:url]'] or not. Requires a siteUrl to be set. Default to true. */

>   fullAbsoluteUrls?: boolean;

Reviewer: ```suggestion
  absoluteSocialMediaPaths?: boolean;
```

Author: they are urls and not pathes?
how about `absoluteSocialMediaUrls`?

Author: used `absoluteSocialMediaUrls`
